txpool-service.md

```diff
@@ -0,0 +1,154 @@
+service attack, figure out how to horizontally scale a solution. This can be achieved by
+creating a stateless edge proxy service that runs a forking EVM. We call this the `op-txpool`.
+
+# Alternatives Considered
+
+## Do No Additional Validation in Mempool
+
+It could be possible to get further than we can think by shipping with no extra validation at all in the mempool.
+This would mean that the extra interop related rules would be validated at block building time only. This can
+create a headache for the block builder and it would take an active adversary to cause problems over time
+as we could block malicious IP addresses or accounts sending transactions that induce denial of service
+at a higher layer.
+
+## Lobby Against 3074
+
+This isn't very prosocial and its likely that something else breaks the "top level call" invariant detection
+in the future anyways.
+
+## Introduce a new EIP to enable our usecase
+
+It is possible to create a new EIP that enables our usecase but it risks going through L1 governance which would
+take a long time and be risky as it could end up being rejected after we invest a lot of time into it.
+
+# Proposed Solution
+
+## `op-txpool`
+
+**This is scoped to the release of interop**
+
+The `op-txpool` service is stateless, can horizontally scale and simply needs to be configured with
+RPC URLs of remote nodes to fetch state from. It runs an RPC server and implements a single method
+in the [txpool](https://geth.ethereum.org/docs/interacting-with-geth/rpc/ns-txpool) namespace:
+
+```
+txpool_validateTransaction(string) -> bool
+```
+
+This RPC method accepts a hex RLP encoded Ethereum transaction and returns whether or not the
+transaction should be considered valid.
+
+The mempool today has both stateless and stateful validation that it performs on every transaction
+as it is entering the mempool.
+
+Stateless:
+- [EIP-2718](https://eips.ethereum.org/EIPS/eip-2718) typing
+- [EIP-3860: Limit and meter initcode](https://eips.ethereum.org/EIPS/eip-3860)
+
+Stateful:
+- Nonce increments based on state or other txs in mempool
+- Balance can pay fee
+- Gas limit not too small and not too large
+- GasFeeCap larger than or equal to basefee
+- 4844 sidecar matches blob hashes
+
+Interop adds a new requirement such that execution is required to know if a transaction entering
+the mempool can be considered valid. This is where `op-txpool` can be very useful - it can execute
+transactions to look for executing messages and then resolve the initiating message from a remote chain.
+This architecture removes the need for the EL client to be aware of the remote chain RPCs.
+It can run optionally, meaning that full nodes don't need to run it. The first iteration of `op-txpool`
+doesn't need to do any of the checking that is done in the mempool already but in the future it can
+also act as a first level of checking before the transactions reach full nodes.
+
+## Implementation
+
```

Comment:
Should add this this needs rate limiting on the number of RPC requests it sends out and also can run with `proxyd` to load balance remote requests against a consistent backend